Content.Server/Anomaly/Effects/ElectricityAnomalySystem.cs

```diff
@@ -27,38 +26,21 @@ public override void Initialize()
         SubscribeLocalEvent<ElectricityAnomalyComponent, AnomalySupercriticalEvent>(OnSupercritical);
     }
 
-    private void OnPulse(EntityUid uid, ElectricityAnomalyComponent component, ref AnomalyPulseEvent args)
+    private void OnPulse(Entity<ElectricityAnomalyComponent> anomaly, ref AnomalyPulseEvent args)
     {
-        var range = component.MaxElectrocuteRange * args.Stability;
-        var xform = Transform(uid);
-        foreach (var (ent, comp) in _lookup.GetEntitiesInRange<MobStateComponent>(xform.MapPosition, range))
-        {
-            _lightning.ShootLightning(uid, ent);
-        }
-    }
+        var range = anomaly.Comp.MaxElectrocuteRange * args.Stability;
 
-    private void OnSupercritical(EntityUid uid, ElectricityAnomalyComponent component, ref AnomalySupercriticalEvent args)
-    {
-        var poweredQuery = GetEntityQuery<ApcPowerReceiverComponent>();
-        var mobQuery = GetEntityQuery<MobThresholdsComponent>();
-        var validEnts = new HashSet<EntityUid>();
-        foreach (var ent in _lookup.GetEntitiesInRange(uid, component.MaxElectrocuteRange * 2))
-        {
-            if (mobQuery.HasComponent(ent))
-                validEnts.Add(ent);
+        var boltCount = (int) (args.Severity * anomaly.Comp.MaxBoltCount);
```

Comment:
instead of this and clamping it, which won't smoothly increase with severity, do this:

`MathHelper.Lerp(anomaly.Comp.MinBoltCount, anomaly.Comp.MaxBoltCount, args.Severity);`

this is just a smooth lerp between the two values (which is how other anomalies handle similar behavior).

Also you can just round this value and cast to an int for the actual count.

Reply:
Done